renku/core/management/workflow/converters/cwl.py

```diff
@@ -367,10 +362,15 @@ def _convert_input(input: CommandInput, basedir: Path):
                 prefix = prefix[:-1]
                 separate = True
 
-        return cwlgen.CommandInputParameter(
-            sanitized_id,
-            param_type=type_,
-            input_binding=cwlgen.CommandLineBinding(position=position, prefix=prefix, separate=separate),
+            input_binding = cwl.CommandLineBinding(position=position, prefix=prefix, separate=separate)
+
+        if position:
+            input_binding = cwl.CommandLineBinding(position=position, prefix=prefix, separate=separate)
+
+        return cwl.CommandInputParameter(
+            id=sanitized_id,
+            type=type_,
+            inputBinding=input_binding,
```

Comment:
note, without this workaround the value of inputBinding in the generated CWL will be `{}` in case `position` and `prefix` are actually `None` that will cause an erroneous execution of the workflow with `cwltool` 🤷

Reply:
`inputBinding=cwl.CommandLineBinding(position=position, prefix=prefix, separate=separate) if position or prefix else None,` would also work, no?

Reply:
yeah, that should fly as well.